In [54]:
# Простейшие операции
# z - комментирование
"""
Это многострочный комментарий
"""
a = 4
b = 2.5
c = a*b
d = "Привет Мир_"

In [22]:
c # Выведем нашу переменную С

10.0

In [23]:
d # Выведем нашу переменную С

'Привет Мир'

In [55]:
# теперь продемонстрируем работу функции
def func_a(arg1, arg2):
    """функция сложения"""
    x = a + b
    return x

In [56]:
func_a(a, b)

6.5

In [30]:
print(d) # функция возвращает изначально строковые значения нашей переменной d

Привет Мир


In [27]:
# типы данных и адрес
type(a) # Выведем тип данных нашей переменной С

# type(b)
# type(c)
# type(d)

int

In [31]:
# Давайте посмотрим что будет если мы сложим переменные a и d
a + d

TypeError: unsupported operand type(s) for +: 'int' and 'str'

In [32]:
a + b # Значит численное значение сложились хорошо

6.5

In [35]:
# а что если мы попробуем сложить переменную D с ней же самой
d + d

'Привет Мир_Привет Мир_'

In [39]:
# Выполним оставшиеся операции с числами которые могут быть использованы к нашим переменным
a - 5 # вполне можем получить отрицательное число

-1

In [40]:
a ** 2 # получим степень числа используя двойной оператор умножения

16

In [41]:
a // b # мы увидим мы увидим операцию с очистительного отделения в котором остаток будет отброшен

1.0

In [42]:
a % b # Давайте наоборот получим остаток от деления

1.5

# Давайте продемонстрируем ещё несколько операций с данными, которые у нас хранятся в переменных

In [48]:
a * d #1 попробуем умножить нашу целочисленную переменную А на переменную со строкой. 
#2 И посмотрим что у нас получилось у нас наша строковая переменная увеличилась в четыре раза.

'Привет Мир_Привет Мир_Привет Мир_Привет Мир_'

In [51]:
# но ещё интересный момент мы можем извлекать символы из наших строковых переменных
d[0] # мы можем обратиться к переменной D содержащую строковое значение по индексу. 
# используем квадратные скобки
# выставим значение 0 поскольку индекс всегда начинается с нуля - получим значение первой буквы

'П'

In [52]:
# а допустим мы ему хотим обратиться к последней букве
d[-1] # тогда используем отрицательный индекс

'_'

In [53]:
d[0:6] #Что если мы хотим выделить наше слово привет Давайте попробуем посчитать количество букв в начале нашей строки. вернёмся вверх и посмотрим Сколько букв у нас было в слове.

'Привет'

# Вопросы для проверки
### 1. Как представлены данные в языке Python?
### Варианты ответов:
    ### а. Представлены в виде объектов
    ### б. Представлены в виде функций
    ### в. Представлены в виде переменных
    ### г. Представлены в виде чисел

# Вопросы для проверки
### 2. Какой тип данных у переменной `a`, если `a = 4`?
### Варианты ответов:
    ### а. int
    ### б. float
    ### в. str
    ### г. list

# Вопросы для проверки
### 3. Что произойдет, если сложить переменные `a` и `d`, где `a = 4`, а `d = "Привет"`?
### Варианты ответов:
    ### а. Числовое значение
    ### б. Строковое значение
    ### в. Произойдет ошибка
    ### г. Результат не определен

# Вопросы для проверки
### 4. Какой тип данных у переменной `b`, если `b = 2.5`?
### Варианты ответов:
    ### а. int
    ### б. float
    ### в. str
    ### г. list

# Вопросы для проверки
### 5. Что вернет функция `type(c)`, если `c = a * b`, где `a = 4`, а `b = 2.5`?
### Варианты ответов:
    ### а. int
    ### б. float
    ### в. str
    ### г. list

# Вопросы для проверки
### 6. Какой результат будет, если выполнить `d * 2`, где `d = "Привет"`?
### Варианты ответов:
    ### а. "ПриветПривет"
    ### б. "Привет"
    ### в. "Привет 2"
    ### г. Произойдет ошибка

# Вопросы для проверки
### 7. Какой результат будет, если выполнить `a // b`, где `a = 4`, а `b = 2.5`?
### Варианты ответов:
    ### а. 1
    ### б. 1.5
    ### в. 2
    ### г. Произойдет ошибка

# Вопросы для проверки
### 8. Какой результат будет, если выполнить `a % b`, где `a = 4`, а `b = 2.5`?
### Варианты ответов:
    ### а. 1
    ### б. 1.5
    ### в. 2
    ### г. Произойдет ошибка

# Вопросы для проверки
### 9. Какой результат будет, если выполнить `a ** 2`, где `a = 4`?
### Варианты ответов:
    ### а. 8
    ### б. 16
    ### в. 64
    ### г. Произойдет ошибка

# Вопросы для проверки
### 10. Какой результат будет, если выполнить `d[1]`, где `d = "Привет"`?
### Варианты ответов:
    ### а. 'П'
    ### б. 'р'
    ### в. 'и'
    ### г. Произойдет ошибка

